sys/net/sock/async/event/sock_async_event.c

```diff
@@ -144,6 +253,12 @@ void sock_udp_event_init(sock_udp_t *sock, event_queue_t *ev_queue,
     ctx->event.cb.udp = handler;
     sock_udp_set_cb(sock, _udp_cb, handler_arg);
 }
+
+void sock_udp_event_close(sock_udp_t *sock)
+{
+    _sock_event_close_common(sock, sock_udp_get_async_ctx(sock), SOCK_ASYNC_UDP);
```

Comment:
There is the small urge to just pass

```suggestion
    _sock_event_close_common(sock, sock_udp_get_async_ctx(sock), (void (*)(void*))sock_udp_close);
```

but it'll probably not save much.

Reply:
Yes I was thinking of doing the same, since all functions take a pointer. Alas, by the C rules, casting functions pointers of different type is UB, no matter what. Although it will probably work on any sane architecture.